src/cmap/handshake/client_metadata.ts

```diff
@@ -156,15 +156,18 @@ export function makeClientMetadata(options: MakeClientMetadataOptions): ClientMe
   return metadataDocument.toObject() as ClientMetadata;
 }
 
-let dockerPromise: Promise<boolean>;
+let isDocker: Promise<void> | boolean | undefined;
 /** @internal */
 async function getContainerMetadata() {
   const containerMetadata: Record<string, any> = {};
-  dockerPromise ??= fs.access('/.dockerenv').then(
-    () => true,
-    () => false
-  );
-  const isDocker = await dockerPromise;
+  if (typeof isDocker !== 'boolean') {
+    try {
+      await (isDocker ??= fs.access('/.dockerenv'));
+      isDocker = true;
+    } catch {
+      isDocker = false;
+    }
+  }
```

Comment:
Can we exempt this one?  This is significantly less readable imo

Reply:
Tried an improvement, lmk what you think